### PR TITLE
build: Set network-passt-binding binary to build statically

### LINKED ISSUE #11586: Passt plugin binary can't work after v1.1.1
report error : /lib64/libc.so.6 version GLIBC 2.34 not found
BYW,release v1.1.1 binary file [kubevirt-passt-binding](https://github.com/kubevirt/kubevirt/releases/download/v1.1.1/kubevirt-passt-binding)
24.1 MB work fine.

### PR DESCRIPTION
**Pull request by brianmcarey**
### What this PR does

Some users have reported issues[1] running the network-passt-binding binary due to glibc conflicts - build the network-passt-binding binary statically for more compatibility

[1] https://github.com/kubevirt/kubevirt/issues/11586

Fixes #11586 

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Special notes for your reviewer
/cc @xpivarc @victortoso 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

File: cmd/sidecars/network-passt-binding/BUILD.bazel [hunk 1]
```diff
@@ -18,6 +18,7 @@ go_library(
 go_binary(
     name = "network-passt-binding",
     embed = [":go_default_library"],
+    static = "on",
```
Comment (vasiliy-ul) [inline cmd/sidecars/network-passt-binding/BUILD.bazel:21]: This binary is run in a container, and I would assume it should use the libraries from that container. Why does it work on some platforms and fails on others? This should be independent of the host OS. Or am I missing smth?

Comment (brianmcarey) [reply cmd/sidecars/network-passt-binding/BUILD.bazel:21]: It is also included as a plain binary in the release assets - https://github.com/kubevirt/kubevirt/releases/tag/v1.1.1

/cc @ormergi  - is the network-passt-bindng binary meant to be included in the release assests? I guess the published artifact is meant to be the image?

Comment (EdDev) [reply cmd/sidecars/network-passt-binding/BUILD.bazel:21]: In the release assets we deliver the custom CNI [1].

https://github.com/kubevirt/kubevirt/tree/main/cmd/cniplugins

So I think @vasiliy-ul has a point here.

Comment (vasiliy-ul) [reply cmd/sidecars/network-passt-binding/BUILD.bazel:21]: Ah, okay, so we have the sidecar binary `network-passt-binding` (this one, which is run in a container) and also a CNI binary `kubevirt-passt-binding` (which is run on the host). Then `kubevirt-passt-binding` needs to be built statically, I assume:

https://github.com/kubevirt/kubevirt/blob/f4aad6e368e8a4fa54fea866939b84a3b27b5512/cmd/cniplugins/passt-binding/cmd/BUILD.bazel#L16-L20

Comment (EdDev) [reply cmd/sidecars/network-passt-binding/BUILD.bazel:21]: Posted https://github.com/kubevirt/kubevirt/pull/11653

Comment (brianmcarey) [reply cmd/sidecars/network-passt-binding/BUILD.bazel:21]: Thanks @EdDev !

**Comment by brianmcarey** [comment]
/close 

Closing in favour of https://github.com/kubevirt/kubevirt/pull/11653

**Pull request closed without merging**